contracts/injective-cosmwasm-mock/src/testing/query_spot_market_test.rs

```diff
@@ -0,0 +1,90 @@
+use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
+use cosmwasm_std::{Addr, Coin};
+use injective_cosmwasm::{checked_address_to_subaccount_id, MarketId};
+use injective_math::{scale::Scaled, FPDecimal};
+use injective_std::types::injective::exchange::v1beta1::{MsgInstantSpotMarketLaunch, QuerySpotMarketsRequest};
+use injective_test_tube::{injective_cosmwasm::SpotMarketResponse, Account, Exchange, InjectiveTestApp, Module, Wasm};
+
+pub const BASE_DENOM: &str = "inj";
+pub const QUOTE_DENOM: &str = "usdt";
+
+pub fn dec_to_proto(val: FPDecimal) -> String {
+    val.scaled(18).to_string()
+}
+
+#[test]
+#[cfg_attr(not(feature = "integration"), ignore)]
+fn test_instantiation() {
+    let app = InjectiveTestApp::new();
+    // create new account with initial funds
+    let accs = app
+        .init_accounts(&[Coin::new(1_000_000_000_000, "usdt"), Coin::new(1_000_000_000_000, "inj")], 2)
+        .unwrap();
+
+    let seller = &accs[0];
+    let buyer = &accs[1];
+    let signer = app.init_account(&[Coin::new(1_000_000_000_000_000_000_000_000_000, "inj")]).unwrap();
+
+    // `Wasm` is the module we use to interact with cosmwasm releated logic on the appchain
+    // it implements `Module` trait which you will see more later.
+    let wasm = Wasm::new(&app);
+    let exchange = Exchange::new(&app);
+
+    // Load compiled wasm bytecode
+    let wasm_byte_code = std::fs::read("../../artifacts/injective_cosmwasm_mock-aarch64.wasm").unwrap();
+    let code_id = wasm.store_code(&wasm_byte_code, None, buyer).unwrap().data.code_id;
+
+    // Instantiate contract
+    let contract_address: String = wasm
+        .instantiate(code_id, &InstantiateMsg {}, Some(&seller.address()), Some("mock-contract"), &[], &seller)
+        .unwrap()
+        .data
+        .address;
+    assert!(contract_address.len() > 0, "Contract address is empty");
+
+    // Execute contract
+    let buyer_subaccount_id = checked_address_to_subaccount_id(&Addr::unchecked(buyer.address()), 1u32);
+    let res = wasm.execute(
+        &contract_address,
+        &ExecuteMsg::TestDepositMsg {
+            subaccount_id: buyer_subaccount_id,
+            amount: Coin::new(100, "usdt"),
+        },
+        &[Coin::new(100, "usdt")],
+        &buyer,
+    );
+    assert!(res.is_ok(), "Execution failed with error: {:?}", res.unwrap_err());
```

Comment:
A more extensive test would also assert the balances, but to be fair, I think it's fine to keep it like this. We don't need to test the logic of the messages, only if an error occurs.